vcl/CMakeLists.txt

```diff
@@ -102,19 +102,38 @@ endif()
 # Install platform-specific compiler detection files
 #
 file(GLOB VXL_COMPILER_DETECTION_FILES ${CMAKE_CURRENT_LIST_DIR}/compilers/*)
-install(FILES ${VXL_COMPILER_DETECTION_FILES}
-      DESTINATION include/vxl/vcl/compilers
-      PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
-      COMPONENT Development )
-install(FILES ${CMAKE_CURRENT_LIST_DIR}/vcl_compiler_detection.h
-      DESTINATION include/vxl/vcl
-      PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
-      COMPONENT Development )
+if("${VXL_INSTALL_INCLUDE_DIR}" STREQUAL "include/vxl")
+  install(FILES ${VXL_COMPILER_DETECTION_FILES}
+        DESTINATION include/vxl/vcl/compilers
+        PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
+        COMPONENT Development )
+  install(FILES ${CMAKE_CURRENT_LIST_DIR}/vcl_compiler_detection.h
+        DESTINATION include/vxl/vcl
+        PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
+        COMPONENT Development )
+else()
+  install(FILES ${VXL_COMPILER_DETECTION_FILES}
+        DESTINATION ${VXL_INSTALL_INCLUDE_DIR}/compilers
+        PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
+        COMPONENT Development )
+  install(FILES ${CMAKE_CURRENT_LIST_DIR}/vcl_compiler_detection.h
+        DESTINATION ${VXL_INSTALL_INCLUDE_DIR}
+        PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
+        COMPONENT Development )
+endif()
 
-vxl_configure_file(${CMAKE_CURRENT_LIST_DIR}/vcl_config_compiler.h.in ${PROJECT_BINARY_DIR}/vcl_config_compiler.h include/vxl/vcl)
-vxl_configure_file(${CMAKE_CURRENT_LIST_DIR}/vcl_config_headers.h.in  ${PROJECT_BINARY_DIR}/vcl_config_headers.h  include/vxl/vcl)
-vxl_configure_file(${CMAKE_CURRENT_LIST_DIR}/vcl_config_manual.h.in   ${PROJECT_BINARY_DIR}/vcl_config_manual.h   include/vxl/vcl)
-vxl_configure_file(${CMAKE_CURRENT_LIST_DIR}/vcl_where_root_dir.h.in  ${PROJECT_BINARY_DIR}/vcl_where_root_dir.h  include/vxl/vcl)
+# If VXL_INSTALL_INCLUDE_DIR is the default value
+if("${VXL_INSTALL_INCLUDE_DIR}" STREQUAL "include/vxl")
```

Comment:
@thewtex 

Can this be condensed:
if()
set(VXL_OVERRIDE_INSTALL_DIR "include/vxl/vcl"
else()
set(VXL_OVERRIDE_INSTALL_DIR ${VXL_INSTALL_INCLUDE_DIR})
endif()
vxl_configure_file(${CMAKE_CURRENT_LIST_DIR}/vcl_config_compiler.h.in ${PROJECT_BINARY_DIR}/vcl_config_compiler.h  ${VXL_OVERRIDE_INSTALL_DIR})